src/detect-flowbits.c

```diff
@@ -105,6 +143,13 @@ static int DetectFlowbitMatchIsset (Packet *p, const DetectFlowbitsData *fd)
 {
     if (p->flow == NULL)
         return 0;
+    if (fd->or_list_size > 0) {
+        for (uint8_t i = 0; i < fd->or_list_size; i++) {
```

Comment:
I think this will work fine if we have 2 or 3 flow bits, but it may be slow if we have many flowbits. Pretty much we walk an array here and per member we walk a list in the flow. In the common case both the list and array will be short, but what if both contain 100 items?

Since the most expensive part is most likely the list walk in the flow, I think it will be more efficient to do something like:
```c
if (fd->or_list_size > 0) {
   return FlowBitIssetFromArray(p->flow, fd->or_list, fd->or_list_size);
}
```
Then `FlowBitIssetFromArray` would walk `f->flowvar` and for each flowbit in that list check the array. I suppose a small and opportunistic optimization could be to pass the minimum and maximum of the id's in the array as well, so you can first check if the flow bit id would possibly be part of the id space. Another option could be to use something like a bloomfilter to do a first quick check.